js/grumbles/edit.controller.js

```diff
@@ -0,0 +1,28 @@
+"use strict";
+
+(function(){
+  angular
+    .module( "grumbles" )
+    .controller( "GrumbleEditController", [
+      "GrumbleFactory",
+      "$stateParams",
+      //"$stateProvider", TODO: Why does this cause errors?
```

Comment:
`$stateProvider` should only be injected into `.config`. I'm a bit surprised that it throws errors -- I thought it just wouldn't do anything -- but perhaps it really is that picky about where it's injected.